paimon-core/src/main/java/org/apache/paimon/privilege/PrivilegeChecker.java

```diff
@@ -24,6 +24,25 @@
 
 /** Check if current user has privilege to perform related operations. */
 public interface PrivilegeChecker extends Serializable {
+    default void assertCanSelectOrInsert(Identifier identifier) {
+        try {
+            assertCanSelect(identifier);
+        } catch (NoPrivilegeException e) {
+            try {
+                assertCanInsert(identifier);
+            } catch (NoPrivilegeException e1) {
+                NoPrivilegeException exception =
+                        new NoPrivilegeException(
+                                e1.getUser(),
+                                e1.getObjectType(),
+                                e1.getIdentifier(),
+                                PrivilegeType.SELECT,
+                                PrivilegeType.INSERT);
+                exception.addSuppressed(e);
+                throw exception;
```

Comment:
```suggestion
                throw new NoPrivilegeException(
                        e1.getUser(),
                        e1.getObjectType(),
                        e1.getIdentifier(),
                        PrivilegeType.SELECT,
                        PrivilegeType.INSERT);
```

Now you only catch `NoProvilegeException`, so the cause of this exception is clear. No need to add suppressed or cause.